src/nbl/video/CVKLogicalDevice.h

```diff
@@ -75,44 +185,137 @@ class CVKLogicalDevice final : public ILogicalDevice
+        vk_createInfo.queueFamilyIndex = familyIndex;
+
+        VkCommandPool vk_commandPool = VK_NULL_HANDLE;
+        if (vkCreateCommandPool(m_vkdev, &vk_createInfo, nullptr, &vk_commandPool) == VK_SUCCESS)
+        {
+            return core::make_smart_refctd_ptr<CVulkanCommandPool>(
+                core::smart_refctd_ptr<CVKLogicalDevice>(this), flags, familyIndex, vk_commandPool);
+        }
+        else
+        {
+            return nullptr;
+        }
     }
             
-    core::smart_refctd_ptr<IDescriptorPool> createDescriptorPool(IDescriptorPool::E_CREATE_FLAGS flags, uint32_t maxSets, uint32_t poolSizeCount, const IDescriptorPool::SDescriptorPoolSize* poolSizes) override
+    core::smart_refctd_ptr<IDescriptorPool> createDescriptorPool(
+        IDescriptorPool::E_CREATE_FLAGS flags, uint32_t maxSets, uint32_t poolSizeCount,
+        const IDescriptorPool::SDescriptorPoolSize* poolSizes) override
     {
-        return nullptr;
+        constexpr uint32_t MAX_DESCRIPTOR_POOL_SIZE_COUNT = 100u;
+
+        assert(poolSizeCount <= MAX_DESCRIPTOR_POOL_SIZE_COUNT);
+
+        // I wonder if I can memcpy the entire array
+        VkDescriptorPoolSize vk_descriptorPoolSizes[MAX_DESCRIPTOR_POOL_SIZE_COUNT];
+        for (uint32_t i = 0u; i < poolSizeCount; ++i)
+        {
+            vk_descriptorPoolSizes[i].type = static_cast<VkDescriptorType>(poolSizes[i].type);
+            vk_descriptorPoolSizes[i].descriptorCount = poolSizes[i].count;
+        }
+
+        VkDescriptorPoolCreateInfo vk_createInfo = { VK_STRUCTURE_TYPE_DESCRIPTOR_POOL_CREATE_INFO };
+        vk_createInfo.pNext = nullptr; // Each pNext member of any structure (including this one) in the pNext chain must be either NULL or a pointer to a valid instance of VkDescriptorPoolInlineUniformBlockCreateInfoEXT or VkMutableDescriptorTypeCreateInfoVALVE
+        vk_createInfo.flags = static_cast<VkDescriptorPoolCreateFlags>(flags);
+        vk_createInfo.maxSets = maxSets;
+        vk_createInfo.poolSizeCount = poolSizeCount;
+        vk_createInfo.pPoolSizes = vk_descriptorPoolSizes;
+
+        VkDescriptorPool vk_descriptorPool;
+        if (vkCreateDescriptorPool(m_vkdev, &vk_createInfo, nullptr, &vk_descriptorPool) == VK_SUCCESS)
+        {
+            return core::make_smart_refctd_ptr<CVulkanDescriptorPool>(
+                core::smart_refctd_ptr<CVKLogicalDevice>(this), vk_descriptorPool);
+        }
+        else
+        {
+            return nullptr;
+        }
     }
             
     core::smart_refctd_ptr<IGPURenderpass> createGPURenderpass(const IGPURenderpass::SCreationParams& params) override
     {
-        return nullptr;
+        // Todo(achal): Hoist creation out of constructor
+        return nullptr; // return core::make_smart_refctd_ptr<CVulkanRenderpass>(this, params);
     }
             
     void flushMappedMemoryRanges(core::SRange<const video::IDriverMemoryAllocation::MappedMemoryRange> ranges) override
```

Comment:
flushing and invalidation are paramount for streaming buffer to work

Reply:
resolved